types/ember-modal-dialog/components.d.ts

```diff
@@ -0,0 +1,6 @@
+declare module 'ember-modal-dialog/components/modal-dialog' {
+    export default class ModalDialog {
+        willDestroyElement(): void
+        didInsertElement(): void
+    }
+}
```

Comment:
should we also have typings for `modal-outlet` and the `modal` service, or do you get the method/type hinting/autocomplete etc for free w/o that? whats the reason for declaring only `willDestroyElement` and `didInsertElement` here?

Reply:
You get them for free. This is because otherwise when you try to use https://github.com/Gavant/gavant-ember-modals/pull/12/files#diff-dc9fb81fb8146ea58a0358fba8189a8aR49 super, it will say `Property 'didInsertElement' does not exist on type 'ModalDialog'`.